xarray/plot/plot.py

```diff
@@ -310,9 +309,14 @@ def line(darray, *args, **kwargs):
 
     # Remove pd.Intervals if contained in xplt.values.
     if _valid_other_type(xplt.values, [pd.Interval]):
-        if interval_step_plot:
+        if kwargs.get('linestyle', '').startswith('steps-'):
             xplt_val, yplt_val = _interval_to_double_bound_points(xplt.values,
                                                                   yplt.values)
+            # just to be sure that matplotlib is not confused
+            kwargs['linestyle'] = kwargs['linestyle'].replace(
```

Comment:
I know this is quite ugly, but does it make sense to import re only for this one line?